testing_and_setup/compass/ocean/global_ocean/scripts/create_E3SM_coupling_files.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python
+"""
+This script creates coupling files needed to run an MPAS-Ocean and
+MPAS-seaice within E3SM.
+
+Load the lastest e3sm-unified conda package.
+
+"""
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+import os
+import shutil
+import subprocess
+from six.moves.configparser import ConfigParser
+import numpy as np
+import glob
+from datetime import datetime
+
+
+def main():
+
+    # obtain configuration settings
+    config = ConfigParser()
+    config.read("create_E3SM_coupling_files.ini")
+    mesh_name = config.get('main', 'mesh_name')
+
+    function_list = ['initial_condition_ocean',
+                     'graph_partition_ocean',
+                     'initial_condition_seaice',
+                     'scrip',
+                     'transects_and_regions',
+                     'mapping_Gcase',
+                     'domain'];
+
+    # create input_data directories
+    make_dir('assembled_files_for_upload/input_data/ocn/mpas-o/' + mesh_name)
+    make_dir('assembled_files_for_upload/input_data/ice/mpas-cice/' + mesh_name)
+    make_dir('assembled_files_for_upload/input_data/cpl/cpl6')
+    make_dir('assembled_files_for_upload/input_data/share/domains')
+
+    success = True
+
+    now = datetime.now()
+    date_string = now.strftime("%y%m%d")
+    for function_name in function_list:
+        print("****** " + function_name + " ******")
+
+        if (config.get(function_name, 'enable').lower() == 'false'):
+            print("Disabled in .ini file")
+        else:
+            currentDir = os.getcwd()
+            make_dir(function_name)
+            os.chdir(function_name)
+    
+            try:
+                globals()[function_name](config, date_string)
```

Comment:
We had discussed an alternative way where `function_list` is a list of the functions themselves.  You could iterate on that list:
```
for function in function_list:
```
and get names like this:
```
function_name = function.__name__
```
and call the function in a more intuitive way:
```
function(config, date_string)
```
Did you decide you prefer this way?  It's an unfamiliar syntax to me and I suspect it wouldn't be very intuitive to other python programmers.

Reply:
I tried, but it sees the function as a string, and gives an error:
```
    function_name = function.__name__
AttributeError: 'str' object has no attribute '__name__'
```
Apparently this doesn't work with this line,
`function_list = ['initial_condition_ocean', ...`
that produces a list of strings.

Reply:
The list of functions needs to be without the quotes:
```
function_list  = [initial_condition_ocean, ...
```

Reply:
Thanks! I did that and it worked.